src/game/mod.rs

```diff
@@ -0,0 +1,68 @@
+pub mod sqlx;
+
+pub mod turn_error;
+pub use turn_error::TurnError;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{Board, Mark, State};
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct Game {
+    board: Board,
+    state: State,
+}
+
+impl Game {
+    pub fn board(&self) -> Board {
+        self.board
+    }
+
+    pub fn state(&self) -> State {
+        self.state
+    }
+
+    pub fn make_turn(&mut self, row: usize, column: usize) -> Result<(), TurnError> {
+        let State::Playing(active_role) = self.state else {
+            return Err(TurnError::GameFinished);
+        };
+
+        if self.board.at(row, column).is_some() {
+            return Err(TurnError::CellOccupied);
+        }
+
+        self.board.mark(row, column, active_role);
+
+        if self.is_won(row, column) {
+            self.state = State::Win(active_role);
+            return Ok(());
+        }
+
+        if self.board.full() {
+            self.state = State::Tie;
+            return Ok(());
+        }
+
+        self.state = match self.state {
+            State::Playing(Mark::Circle) => State::Playing(Mark::Cross),
+            State::Playing(Mark::Cross) => State::Playing(Mark::Circle),
+            other => other,
+        };
+
+        Ok(())
+    }
+
+    fn is_won(&self, row: usize, column: usize) -> bool {
+        let row_win = self.board.at(row, 0) == self.board.at(row, 1)
+            && self.board.at(row, 1) == self.board.at(row, 2);
```

Comment:
You should not hardcode board size like this (use loop instead), if you ever expect to use different numbers (not only 3) in BOARD_SIZE. Also you could make early return and not calculate other win conditions if this condition is true.